File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    let answer = 0;
    
    function Move(storey) {
        if(storey === 0) {
            return;
        }

        if(Math.floor(storey % 10) > 5 || (Math.floor(storey % 10) === 5) && Math.floor(storey % 100) / 10 >= 5) {
            answer += 10 - Math.floor(storey % 10);
            storey += 10;
        }
        else {
            answer += Math.floor(storey % 10);
        }
        
        Move(Math.floor(storey / 10));
    }
    
    Move(storey);
    
    return answer;
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.cpp
#include <string>
#include <cstring>
#include <vector>

using namespace std;

int answer = 0;

void Calcul(int num) {
    if(num == 0)
        return;
    
    if(num % 10 > 5 || (num % 10 == 5 && (num % 100) / 10 >= 5)) {
        answer += 10 - (num % 10);
        num += 10;
    }
    else {
        answer += num % 10;
    }
    
    Calcul(num / 10);
}

int solution(int storey) {
    Calcul(storey);
    
    return answer;
}